cdap-integration-test/src/test/java/io/cdap/cdap/test/IntegrationTestBaseTest.java

```diff
@@ -65,12 +68,18 @@ public void testDeployApplicationInNamespace() throws Exception {
 
     ApplicationClient applicationClient = new ApplicationClient(clientConfig);
     Assert.assertEquals(AllProgramsApp.NAME, applicationClient.list(namespace).get(0).getName());
-    applicationClient.delete(namespace.app(AllProgramsApp.NAME));
+    ApplicationDetail appDetail = applicationClient.get(new ApplicationId(namespace.getNamespace(),
+                                                                          AllProgramsApp.NAME));
+    applicationClient.delete(namespace.app(AllProgramsApp.NAME, appDetail.getAppVersion()));
```

Comment:
Do we provide a way to delete specific version of an app? I believe it deletes all versions, so version should not be needed here

Reply:
This one is deleting a specific version IIRC @sanjanasandeep

Reply:
yes^

Reply:
Do we generally allow to delete a specific version? If so, please add tests that user can delete latest and non-latest versions of the app

Reply:
yes added this test in  `testDelete` in `AppLifecycleHttpHandlerTest`